truffle/src/main/java/org/jruby/truffle/language/SnippetNode.java

```diff
@@ -88,13 +101,8 @@ public Object execute(VirtualFrame frame, Object... arguments) {
                 parentFrameArguments,
                 frameDescriptor);
 
-        if (arguments.length != parameterFrameSlots.length) {
-            CompilerDirectives.transferToInterpreter();
-            throw new UnsupportedOperationException("number of arguments doesn't match number of parameters");
-        }
-
         for (int n = 0; n < parameters.length; n++) {
-            parentFrame.setObject(parameterFrameSlots[n], arguments[n]);
+            parentFrame.setObject(parameterFrameSlots[n], arguments[2 * n + 1]);
         }
```

Comment:
Can we extract this to a separate method and only annotate that one with ExplodeLoop?